packages/components/addon/components/hds/code-block/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { htmlSafe } from '@ember/template';
+import { guidFor } from '@ember/object/internals';
+
+import Prism from 'prismjs';
+
+import 'prismjs/plugins/line-numbers/prism-line-numbers';
+import 'prismjs/plugins/line-highlight/prism-line-highlight';
+
+import 'prismjs/components/prism-bash';
+import 'prismjs/components/prism-go';
+import 'prismjs/components/prism-markup-templating';
+import 'prismjs/components/prism-handlebars';
+import 'prismjs/components/prism-hcl';
+import 'prismjs/components/prism-json';
+import 'prismjs/components/prism-log';
+import 'prismjs/components/prism-shell-session';
+import 'prismjs/components/prism-yaml';
+
+export default class HdsCodeBlockIndexComponent extends Component {
+  @tracked prismCode = '';
+
+  /**
+   * Generates a unique ID for the code content
+   *
+   * @param preCodeId
+   */
+  preCodeId = 'pre-code-' + guidFor(this);
+
+  /**
+   * @param code
+   * @type {string}
+   * @description code text content for the CodeBlock
+   */
+  get code() {
+    const code = this.args.value;
+
+    assert(
+      '@code for "Hds::CodeBlock" must have a valid value',
+      code !== undefined
+    );
+
+    if (Prism?.plugins?.NormalizeWhitespace) {
+      return Prism.plugins.NormalizeWhitespace.normalize(code);
+    }
+
+    return code;
+  }
+
+  /**
+   * @param language
+   * @type {string}
+   * @default undefined
+   * @description name of coding language used within CodeBlock for syntax highlighting
+   */
+  get language() {
+    return this.args.language ?? undefined;
+  }
+
+  /**
+   * @param hasLineNumbers
+   * @type {boolean}
+   * @default true
+   * @description Displays line numbers if true
+   */
+  get hasLineNumbers() {
+    return this.args.hasLineNumbers ?? true;
+  }
+
+  /**
+   * @param isStandalone
+   * @type {boolean}
+   * @default true
+   * @description Make CodeBlock container corners appear rounded
+   */
+  get isStandalone() {
+    return this.args.isStandalone ?? true;
+  }
+
+  /**
+   * @param hasLineWrapping
+   * @type {boolean}
+   * @default false
+   * @description Make text content wrap on multiple lines
+   */
+  get hasLineWrapping() {
+    return this.args.hasLineWrapping ?? false;
+  }
+
+  @action
+  setPrismCode(element) {
+    const code = this.code;
+    const language = this.language;
+    const grammar = Prism.languages[language];
+
+    if (code && language && grammar) {
+      this.prismCode = htmlSafe(Prism.highlight(code, grammar, language));
+    } else {
+      this.prismCode = htmlSafe(Prism.util.encode(code));
```

Comment:
> but noting code is required

on that case the `if (code &&`  can have the `code` part removed)

Reply:
> (as it was not designed for empty content).

@alex-ju you mean _designed_ or _implemented_? I am asking because your comment made me think: what if in some (rare, but possible) case the code is lazy loaded or generated at the next cycle after the component rendering? in that case for a short amount of time, the component could be in a state in which `@value` is undefined. Or should we ask consumer to provide always a value (eg. an empty string)? Thinking out loud of "what if".

Reply:
What I meant by that is that if the component is lacking content the height of the code container is slightly smaller than when there's a line of code and there are no line numbers.